test/parallel/test-sqlite-backup.mjs

```diff
@@ -0,0 +1,226 @@
+import '../common/index.mjs';
+import tmpdir from '../common/tmpdir.js';
+import { join } from 'node:path';
+import { backup, DatabaseSync } from 'node:sqlite';
+import { describe, test } from 'node:test';
+import { writeFileSync } from 'node:fs';
+
+let cnt = 0;
+
+tmpdir.refresh();
+
+function nextDb() {
+  return join(tmpdir.path, `database-${cnt++}.db`);
+}
+
+function makeSourceDb() {
+  const database = new DatabaseSync(':memory:');
+
+  database.exec(`
+    CREATE TABLE data(
+      key INTEGER PRIMARY KEY,
+      value TEXT
+    ) STRICT
+  `);
+
+  const insert = database.prepare('INSERT INTO data (key, value) VALUES (?, ?)');
+
+  for (let i = 1; i <= 2; i++) {
+    insert.run(i, `value-${i}`);
+  }
+
+  return database;
+}
+
+describe('backup()', () => {
+  test('throws if the source database is not provided', (t) => {
+    t.assert.throws(() => {
+      backup();
+    }, {
+      code: 'ERR_INVALID_ARG_TYPE',
+      message: 'The "sourceDb" argument must be an object.'
+    });
+  });
+
+  test('throws if path is not a string', (t) => {
+    const database = makeSourceDb();
+
+    t.assert.throws(() => {
+      backup(database);
```

Comment:
Per web platform API conventions, APIs that return a Promise should generally reject the promsie as opposed to throwing synchronously. I know we haven't always been super consistent with that for Node.js APIs tho. Non-blocking but I think I would prefer this to reject rather than throw.

Reply:
Isn't this an input validation issue though? Node APIs generally throw in this case.

Reply:
Yep, that's why it's not blocking and just a preference.